configure.ac

```diff
@@ -729,7 +729,7 @@ libdovecot_headers=
 libdovecot_c_files=
 non_libdovecot_headers=
 non_libdovecot_c_files=
-all_files=`find $srcdir/src -name '*.[[ch]]' | grep -v '/src/config/all-settings.c' | grep -v '/src/lib-settings/' | grep -v '/test-' | xargs grep '\\(struct setting_parser_info [[a-z]]\\)\\|\\(struct service_settings [[a-z]]\\)\\|\\(<settings checks>\\)' | sed 's/:.*//' | sort | uniq`
+all_files=`find $srcdir/src -name '*.[[ch]]' | grep -v '/src/config/all-settings.c' | grep -v '/src/lib-settings/' | grep -v '/test-' | xargs egrep '(struct setting_parser_info [[a-z]])|(struct service_settings [[a-z]])|(<settings checks>)' | sed 's/:.*//' | sort | uniq`
```

Comment:
I remember reading somewhere egrep is going to be deprecated. Should use grep -E instead.

Reply:
Indeed, `egrep` was removed from POSIX: https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_xcu_chap04.html

Reply:
After some thinking the cleaner way is using `$GREP`, `$EGREP` and `$SED` and allow configure to decided which should be used on this platform.

Reply:
I found that using $SED seems to break something else.

Reply:
it seems that For Whatever Reason, it does not correctly collect all dovecot la files in linux.

Reply:
I may have a look on it in hour or two

Reply:
@cmouse may I ask you to share an example that doesn't correctly collected on Linux after this changes?

Reply:
I have no idea, we solved it differently in main (please see 6a337ecd25dfc583ff3f16c8f448a2846d0db261) and the $SED change caused, at least for me, that e.g. lib-lua/libdlua.la was missing from lib-dovecot.

Reply:
Thus, I haven't see any changes in `src/config/all-settings.c` before this commit, and after that on ubuntu 24.04

Reply:
@cmouse I had removed `$SED` part and seems that it generates everything that it needs on OpenBSD.